endpoints/_includes/functions.php

```diff
@@ -746,6 +748,32 @@ function helplineSearch($latitude, $longitude)
     return json_decode(get($search_url));
 }
 
+function getBMLTServer($latitude, $longitude, $search_type)
+{
+    $helpline_search_radius = setting('helpline_search_radius');
+    if ($search_type == SearchType::VOLUNTEERS) {
+        if (setting('tomato_helpline_fallback')) {
+            $bmlt_search_endpoint = setting('tomato_url') . "/client_interface/json/?switcher=GetSearchResults&data_field_key=root_server_uri&sort_results_by_distance=1&long_val={LONGITUDE}&lat_val={LATITUDE}&geo_width=" . $helpline_search_radius;
+            $search_url = str_replace("{LONGITUDE}", $longitude, str_replace("{LATITUDE}", $latitude, $bmlt_search_endpoint));
+            $search_results = json_decode(get($search_url));
+            $pinged_BMLT_root = $search_results[0] -> root_server_uri;
+            $bmlt_root = $GLOBALS['bmlt_root_server']."/";
+            if ($pinged_BMLT_root != $bmlt_root) { // If the rootserver handling the phonecall is not the local rootserver, enable tomato routing - otherwise handle call routing locally.
+                $GLOBALS['settings_whitelist']['tomato_helpline_routing']['default'] = true;
```

Comment:
Btw, this isn't going to change anything, I do not recommend modifying the behavior of settings in this manner.

Reply:
Changing the tomato_helpline_routing from within the instance allows it to simply flip a switch that the rest or the already built code can work with. Keeps it unobtrusive to the existing code base. Would require more extensive rewriting to otherwise accomplish the same thing.

So would this mean that even once the above requests are satisfied this will be rejected for the menthod? Or are you saying this won’t work? It has tested out working for all possible logic conditions.

Reply:
The problem with this is that it's not embedded in the session, so the behavior will be lost on subsequent calls if any.  This is not a code pattern that exists anywhere else and may cause unwanted side effects.

Reply:
It is intentionally embedded in the session - it is a temporary flipping of the switch for the purpose of the session it is inside and all references to tomato_helpline_routing - so that no other code needed to be modified. It only flips that switch under the following conditions : volunteer search + tomato_helpline_fallback true + compare target rootserver vs config rootserver is different. For all 3 of those to be true is a singular set of conditions that prevents interference with other functions.